cpp/src/Ice/OutgoingAsync.cpp

```diff
@@ -604,15 +618,15 @@ ProxyOutgoingAsyncBase::runTimerTask()
 
 OutgoingAsync::OutgoingAsync(ObjectPrx proxy, bool synchronous)
     : ProxyOutgoingAsyncBase(std::move(proxy)),
-      _encoding(getCompatibleEncoding(_proxy->_getReference()->getEncoding())),
```

Comment:
What was this `getCompatibleEncoding` for?

Reply:
https://github.com/zeroc-ice/ice/blob/f76d05d208a91ad93bd71fdce3558cb38b92c397/cpp/include/Ice/Protocol.h#L233

Some weird "adaptation" for unknown encodings. For example, if the proxy said "encoding 1.3" but the highest encoding we support is 1.1, we would switch to encoding 1.1 when making the invocation. With my changes, we just fail in this situation.